aldryn_newsblog/search_indexes.py

```diff
@@ -30,11 +33,17 @@ def index_queryset(self, using=None):
         return qs
 
     def get_index_queryset(self, language):
-        return self.get_model().objects.published().active_translations(
-            language_code=language).filter(app_config__search_indexed=True)
+        return (
+            self.get_model().objects
+                            .published()
+                            .language(language)
+                            .active_translations(language_code=language)
+                            .filter(app_config__search_indexed=True)
+        )
```

Comment:
`get_index_queryset` do not apply filters on translated fields, it is done on `index_queryset` and there we user `translated`.